src/components/utils/LayerProvider.tsx

```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import {LayerManager} from './LayerManager';
+
+const LayerContext = React.createContext(new LayerManager());
+
+export type LayerProviderProps = {
+    container: React.RefObject<HTMLElement>;
+};
+
+export const LayerProvider: React.FC<LayerProviderProps> = ({container, children}) => {
+    const context = React.useMemo(() => {
+        return new LayerManager(container);
```

Comment:
each invocation will (in theory) create new instance of LayerManager

it is better to explicitly add some kind of `destroy` method and invoke it on ref change

Reply:
creating another LayerManager instance in same LayerProvider is very rare. so i don't want complicate code for this.

Reply:
That mean you will pass through memory leak, since `window` listeners is not cleared